api/src/test/java/org/openmrs/api/ObsServiceTest.java

```diff
@@ -1998,4 +2003,99 @@ public void shouldNotAutomaticallySetStatusWhenManuallyCopyingAnObs() throws Exc
 		assertThat(existing.getVoided(), is(true));
 		assertThat(newObs.getStatus(), is(Obs.Status.FINAL));
 	}
+
+	/**
+	 * @see ObsService#getObsReferenceRangeById(Integer)
+	 */
+	@Test
+	public void getObsReferenceRangeById_shouldReturnObsReferenceRange() {
+		Obs obs = buildObservation();
+
+		obsService.saveObs(obs, null);
+
+		ObsService obsService = Context.getObsService();
+		ObsReferenceRange fetchedReferenceRange = obsService.getObsReferenceRangeById(obs.getObsReferenceRange().getId());
+
+		assertEquals(obs.getObsReferenceRange().getId(), fetchedReferenceRange.getId());
+	}
+
+	/**
+	 * @see ObsService#getLatestObsByConceptId(String)
+	 */
+	@Test
+	public void getLatestObsByConceptId_shouldReturnLatestObs() throws InterruptedException {
+		ObsService obsService = Context.getObsService();
+		
+		Obs obs1 = buildObservation();
+		obs1.setComment("First Obs");
+		obsService.saveObs(obs1, null);
+
+		// Introduce a delay to ensure a different timestamp for the second obs
+		Thread.sleep(1000);
+		
+		Obs obs2 = buildObservation();
+		obs2.setComment("Second Obs");
+		obsService.saveObs(obs2, null);
+		
+		Obs latestObs = obsService.getLatestObsByConceptId("4089");
+		assertNotNull(latestObs);
+		assertEquals("Second Obs", latestObs.getComment());
+	}
+
+	@Test
+	public void saveObsReferenceRange_shouldSaveReferenceRangeAfterSavingObs() {
+		Obs obs = buildObservation();
+		obs.setUuid("ace759d9-d126-4a3f-ae1a-cad4c132f38b");
```

Comment:
You do not have to set the uuid because it will be automatically assigned.

Reply:
I needed it here so that I can use it to get the saved Obs by uuid.

Reply:
Since you call `obs.getUuid()`, it means that the automatically assigned value would work. Not so?